misc/python/materialize/mzcompose/services.py

```diff
@@ -100,8 +100,8 @@ def __init__(
 
         if system_parameter_defaults is None:
             system_parameter_defaults = {
-                "enable_upsert_source_disk": "true",
-                "upsert_source_disk_default": "true",
+                "enable_disk_cluster_replicas": "true",
+                "disk_cluster_replicas_default": "true",
```

Comment:
This seems surprising; do we really want to be testing disk replicas by default?

Reply:
It only affects upsert sources, which matches the previous behavior

Reply:
Right, I think the previous behavior was surprising too!

Reply:
This happened here: https://github.com/MaterializeInc/materialize/pull/19371

Reply:
I'm somewhat comforted by the fact that this is what @philip-stoev wants, but I'm still spooked by this. Doesn't this remove almost all test coverage of _non_-disk upsert sources?

Reply:
@benesch your concerns are true. My decision to get the CI to run with disk-based sources by default was based on the assumption that DISK-based sources are rapidly going to become **the** one and only default for all of our customers as the existing memory-only situation is not really sustainable for anything but the smallest demos. If this assumption was incorrect, we could switch things around:
- revert the CI default so that it is back to OFF
- have just a Zippy run with disk-based sources ON

Or, we could keep the default of disk being ON for the entire CI and have a Zippy job that exercises the OFF situation (which will happen once https://github.com/MaterializeInc/materialize/issues/19627 is implemented).

Your verdict will be much appreciated as so far I have been guided by my gut feeling only and there is probably a better way to decide than me looking at the entrails of sacrificed birds.

Same conundrum applies for MFP pushdown (and for any other feature flag, for that matter). I am setting some flags to be default in the CI on the assumption that they will rapidly become ON as a default for all of our customers.

Reply:
The cause of the confusion here is that we've waffled on the deployment strategy for this feature substantially. At first we were expecting that 
we'd be able to enable disk for all sources unconditionally. But AWS doesn't have enough i4i instances to make that viable.

So instead we're pushing the decision onto the customer. Each customer has to choose for each cluster whether they want disk. We don't yet know how many customers are going to choose to enable disk. And the percentage of disk-backed sources will be influenced substantially by whether we choose to charge more for disk-backed instances or not.

In the next month or two, we're going to enable this feature for only a few choice customers. We're going to ask those customers to run disk sources _in parallel_ to their existing sources. Once we've validated that the feature is working well, we'll tackle the question of whether we should charge more for disk-backed clusters.

So I'd say the best course of action here _for now_ is to leave the feature off by default in CI, and enable it only for a few choice tests. We should revisit changing the default after this initial "private preview" phase, when we know whether we'll be charging extra for disk sources or not.